LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/EC/EmbeddedController.cs

```diff
@@ -302,6 +312,19 @@ public abstract class EmbeddedController : Hardware
                 { ECSensor.FanCPUOpt, new EmbeddedControllerSource("CPU Optional Fan", SensorType.Fan, 0x00b0, 2) }
             }
         },
+        {
+            BoardFamily.Intel400, new Dictionary<ECSensor, EmbeddedControllerSource>
+            {
+                { ECSensor.TempVrm, new EmbeddedControllerSource("VRM", SensorType.Temperature, 0x003e) },
+                { ECSensor.TempChipset, new EmbeddedControllerSource("Chipset", SensorType.Temperature, 0x003a) },
+                { ECSensor.TempTSensor, new EmbeddedControllerSource("T Sensor", SensorType.Temperature, 0x003d, blank: -40) },
+                { ECSensor.TempWaterIn, new EmbeddedControllerSource("Water In", SensorType.Temperature, 0x0100, blank: -40) },
+                { ECSensor.TempWaterOut, new EmbeddedControllerSource("Water Out", SensorType.Temperature, 0x0101, blank: -40) },
+                { ECSensor.FanCPUOpt, new EmbeddedControllerSource("CPU Optional Fan", SensorType.Fan, 0x00b0, 2) },
+                { ECSensor.CurrCPU, new EmbeddedControllerSource("CPU", SensorType.Current, 0x00f4) }, // todo: needs to be checked, value becomes negative after a certain threshold
```

Comment:
Have you checked?

Reply:
Unfortunately I have no idea why it happens, the value at idle is correct but whenever I put a relatively important load on the CPU it becomes negative (and is of course completely wrong), so I don't know, should I make a new pr without this line of code?

Reply:
@PhyxionNL I'm awaiting your reply, thanks

Reply:
You can slash the broken line out first. Perhaps @zeule knows what's happening?

Reply:
The value is unsigned, we read it a signed. Thanks for spotting the bug!

Reply:
@zeule how do I fix it? for now i've just removed the line, I await instructions from you

Reply:
@Shadotty before making changes, could you, please, confirm that that's indeed that case by observing at what value it flips? It should be at 128 if that's caused by the signed/unsigned mismatch.

Reply:
I will check as soon as I can, it's difficult to get a 128A load but i will try

Reply:
@zeule yes, I can confirm that the value becomes negative after exceeding 128A, so, what should be done here to fix this?

Reply:
Thank you for testing that! I guess we can safely assume that this property is the same for all models, and them something like this should do the trick:
```diff
diff --git a/LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/EC/EmbeddedController.cs b/LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/EC/EmbeddedController.cs
index dc3399a..0e9b111 100644
--- a/LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/EC/EmbeddedController.cs
+++ b/LibreHardwareMonitorLib/Hardware/Motherboard/Lpc/EC/EmbeddedController.cs
@@ -396,7 +396,7 @@ public abstract class EmbeddedController : Hardware
         {
             int val = _sources[si].Size switch
             {
-                1 => unchecked((sbyte)_data[readRegister]),
+                1 => _sources[si].Type switch { SensorType.Temperature => unchecked((sbyte)_data[readRegister]), _ => _data[readRegister] },
                 2 => unchecked((short)((_data[readRegister] << 8) + _data[readRegister + 1])),
                 _ => 0
             };
```

Reply:
Turns out the Linux driver uses exactly this strategy for two years already (https://github.com/torvalds/linux/commit/8aba9ca62677570abf8c6b62611adb85bf1580ca), discovered in the same way (through CPU current), and I forgot to bring the change here. My bad…

Reply:
So can I go ahead and add the line back to the code?

Reply:
But of course.

Reply:
Done, I'll wait for approval!